pkg/route/validation/validation_test.go

```diff
@@ -843,6 +978,717 @@ func TestValidateRoute(t *testing.T) {
+		},
+		{
+			name: "Should give an error if the header value exceeds 16384 chars",
+			route: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "wildcardpolicy",
+					Namespace: "foo",
+				},
+				Spec: routev1.RouteSpec{
+					Host:           "subdomain.wildcard.test",
+					To:             createRouteSpecTo("serviceName", "Service"),
+					WildcardPolicy: routev1.WildcardPolicySubdomain,
+					HTTPHeaders: &routev1.RouteHTTPHeaders{
+						Actions: routev1.RouteHTTPHeaderActions{
+							Response: []routev1.RouteHTTPHeader{
+								{
+									Name: "X-SSL-Client-Cert",
+									Action: routev1.RouteHTTPHeaderActionUnion{
+										Type: routev1.Set,
+										Set: &routev1.RouteSetHTTPHeader{
+											Value: tooLargeValue,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedErrorMessage: fmt.Sprintf("spec.httpHeaders.actions.response[0].action.set.value: Invalid value: %q: value exceeds the maximum length, which is 16384", tooLargeValue),
+		},
+		{
+			name: "Should give an error if the header value is 0 chars",
+			route: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "wildcardpolicy",
+					Namespace: "foo",
+				},
+				Spec: routev1.RouteSpec{
+					Host:           "subdomain.wildcard.test",
+					To:             createRouteSpecTo("serviceName", "Service"),
+					WildcardPolicy: routev1.WildcardPolicySubdomain,
+					HTTPHeaders: &routev1.RouteHTTPHeaders{
+						Actions: routev1.RouteHTTPHeaderActions{
+							Response: []routev1.RouteHTTPHeader{
+								{
+									Name: "X-SSL-Client-Cert",
+									Action: routev1.RouteHTTPHeaderActionUnion{
+										Type: routev1.Set,
+										Set: &routev1.RouteSetHTTPHeader{
+											Value: "",
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedErrorMessage: "spec.httpHeaders.actions.response[0].action.set.value: Required value",
```

Comment:
is this correct?  I thought the error message change to include the header name for a value that was missing?

Reply:
From my understanding this is a correct error message.

Reply:
added header name

Reply:
added headerName

Reply:
We lost the header name in the latest commit.

Reply:
Sorry, @candita, we should have discussed this before the change was pushed.  Anyway, my opinion is that including the header name is (a) inconsistent with other validation and (b) misleading because it implies that the specific header `x-ssl-client-cert` requires a value when in fact a value is _always_ required for _any_ header.  I don't feel very strongly about this, though, so if you still prefer to have the header name in the error message, I'll defer to you.